server/src/streem/util.ts

```diff
@@ -0,0 +1,33 @@
+import { Request } from 'express';
+import { createHmac, timingSafeEqual } from 'crypto';
+import { SIGNING_KEY_SECRET } from '../routes/webhooks';
+
+export function validateWebhookRequest(req: Request): Boolean {
+    const hmac = createHmac('sha256', Buffer.from(SIGNING_KEY_SECRET, 'utf-8').toString());
+
+    // make sure the request is current
+    const sent_at = req.headers['streem-sent-at']! as string;
+    const req_time_skew = Date.now() - Date.parse(sent_at)
+    const allowed_skew = 500 // adjust timedelta (in ms) as desired
+    if (req_time_skew > allowed_skew) {
+        return false; // request is too far in the past 
+    } else if (req_time_skew < -allowed_skew) {
+        return false; // request is in the future
+    }
+
+    // calculate the signature input message
+    const header_names = (req.headers['streem-signature-headers']! as string).split(':')
+    const headers: string[] = header_names.map(header_name => {
+        return `${header_name}=${req.headers[header_name.toLowerCase()]!}`;
```

Comment:
No, we shouldn't assume what case they'll be. The validation needs to compute the signature using the casing for each header name as it is listed in the `Streem-Signature-Headers` header (even if that is different from the casing of the header in the `req` object, since that is sometimes modified by middleware). From our design doc:

> Extract the name and value of each header listed in Streem-Signature-Headers from the HTTP request. Make sure to preserve the order of headers as listed in Streem-Signature-Headers, as well as the capitalization of the header names, when computing the signature below.

Reply:
Which actually means that (I think) Ghislain's code here is correct

Reply:
That's right. I looked at the raw payload and noticed we were using lowercase. E.g., 
``` 
  'streem-sent-at': '2023-09-28T22:21:07.555053Z',
  'streem-signature': '27bc2a71b737f98c1d529f8133f5b9d56bae71c39f2b1f95f020c42b8667e691, f25df92897c9a556689a3d610af1e50fa0a2e26945dc2349a28365c137702612',
  'streem-signature-headers': 'Streem-Sent-At',
```